src/scripts/components/SmartTableHeader.jsx

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  Icon,
+} from '.';
+
+class SmartTableHeader extends React.Component {
+  state = {
+    sortKey: this.props.sortKey,
+    activeCaretClassName: 'u-text-body',
+    headers: this.props.headers,
+  };
+
+  headerHandler = (headerClicked) => {
+    const { headers } = this.state;
+    const oldDirection = this.state.headers[headerClicked].direction;
+    const lastActiveHeader = Object.keys(headers).find(header => (headers[header].default === true));
```

Comment:
Is there a better way to manage the properties of the headers so it is clear how to work with them?

1. `default` isn't very descriptive, so it can bee hard to understand what this bool is doing.
2. Direction represented as ints will also create confusion for devs using this component. 

I'm not sure what the best answer is, but it'd be easier to work with this in the future if our direction was a `oneOf` property with values of `ascending` and `descending` the component can handle parsing this data to power the sorting.

Not sure about a better name for `default` let's chat about this one.

Reply:
Yeah we can do this as `ascending` instead of default.